### PR TITLE
Merge getRuntimeLibraryPath and getRuntimeStaticLibraryPath

### PR DESCRIPTION
**Pull request by lanza**
These two functions are duplicates except for one string. Merge them and
add an argument for whether we want static or shared.

### CODE REVIEW

**Comment by lanza** [comment]
@compnerd smoke test and merge please

File: lib/Driver/ToolChains.cpp [hunk 1]
```diff
@@ -1037,7 +1038,11 @@ static void getRuntimeLibraryPath(SmallVectorImpl<char> &runtimeLibPath,
     runtimeLibPath.append(programPath.begin(), programPath.end());
     llvm::sys::path::remove_filename(runtimeLibPath); // remove /swift
     llvm::sys::path::remove_filename(runtimeLibPath); // remove /bin
-    llvm::sys::path::append(runtimeLibPath, "lib", "swift");
+    if (shared) {
+      llvm::sys::path::append(runtimeLibPath, "lib", "swift");
+    } else {
+      llvm::sys::path::append(runtimeLibPath, "lib", "swift_static");
+    }
```
Comment (compnerd) [inline lib/Driver/ToolChains.cpp]: I think this is better as a ternary:

    llvm::sys::path::append(runtimeLibpath, "lib", shared ? "swift" : "swift_static");

File: lib/Driver/ToolChains.cpp [hunk 2]
```diff
@@ -1047,43 +1052,21 @@ static void getClangLibraryPath(const ToolChain &TC, const ArgList &Args,
                                 SmallString<128> &LibPath) {
   const llvm::Triple &T = TC.getTriple();
 
-  getRuntimeLibraryPath(LibPath, Args, TC);
+  getRuntimeLibraryPath(LibPath, Args, TC, true);
```
Comment (compnerd) [inline lib/Driver/ToolChains.cpp]: Please add a comment:

    getRuntimeLibraryPath(LibPath, Args, TC, /*Shared=*/true);

Similar throughout.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: lib/Driver/ToolChains.cpp (hunk 2 above)
Comment (lanza) [reply lib/Driver/ToolChains.cpp]: Gross, if only there was a language with functions with named arguments.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Pull request merged**